### PR TITLE
feat: attachFile string shorthands

### LINKED ISSUE #18753: cy.attachFile for aliases, fixtures and file paths
### What would you like?

As a follow-up to https://github.com/cypress-io/cypress/issues/18752, we'll expand the features `cy.attachFile()` supports for increased usability, allowing developers to load their data in a wider variety of ways.

This ticket will add support for:
```js
cy.attachFile('path/to/myFile.json')
cy.attachFile('@myAlias')
```

The details of this feature are described more thoroughly in our internal documents, and will be described here (and in the documentation) when we start work on them.

### Why is this needed?

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by BlueWinds**
- Closes #18753

### User facing changelog
A full user-facing changelog will be included when PRing from the feature branch into develop / a release.

This PR adds the ability to use string shorthands with attachFile:

```
cy.attachFile('path/to/myFile.json')
cy.attachFile('@myAlias')
```

### PR Tasks
- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? No, documentation will be written up separately.
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)? Types were written with support for strings from the beginning, so no changes needed there.

### CODE REVIEW

File: packages/driver/src/cypress/error_messages.ts [hunk 1]
```diff
@@ -134,23 +134,27 @@ export default {
   attachFile: {
     invalid_action: {
       message: `${cmd('attachFile')} \`action\` can only be \`input\` or \`drag-n-drop\`. You passed: \`{{action}}\`.`,
-      docsUrl: 'https://on.cypress.io/attachFile',
+      docsUrl: 'https://on.cypress.io/attachfile',
```
Comment (flotwig) [inline packages/driver/src/cypress/error_messages.ts:137]: fyi, you can also declare the `docsUrl` higher up on this tree and have it be the default for the errors, see `net_stubbing.docsUrl` for an example of this inheritance

File: packages/driver/src/cy/commands/actions/attach-file-tech-brief/README.md [hunk 2]
```diff
@@ -42,21 +41,21 @@ This triggers `input` and `change` events on the given input element, signaling
...
-    filePath: '/imaginary/path/to/file.txt',
+    fileName: '/imaginary/path/to/file.txt',
     lastModified: Date.now(),
     mimeType: 'application/octet-stream',
   })
 ```
 
-- `contents` can be any of the above formats - `fixture:file.json`, an `@alias`, or a `Buffer()`. Required.
-- `filePath` is a string. If undefined, it defaults to the actual path (if we have one from `cy.readFile()` or `cy.fixture()`), or an empty string if none can be inferred.
+- `contents` can be any of the above formats - `file.json`, an `@alias`, a `Buffer()`, or an object to JSON.stringify(). Required.
```
Comment (flotwig) [inline packages/driver/src/cy/commands/actions/attach-file-tech-brief/README.md:57]: I see why you want to re-use `contents` for multiple different meaning, even when it's in an object, I think. Is it so that `cy.attachFile(contents)` and `cy.attachFile({ contents })` work exactly the same?

Option B would've been to let the user use separate, mutually-exclusive keys in this object... `fileName`, `alias`, `contents` (which could still work with strings or Buffers or objects). This would be a little easier to TSify and debug, I think.

Can you comment on why the API was designed this way instead of option B?

**Review by flotwig — APPROVED** [review]

File: packages/driver/src/cy/commands/actions/attach-file-tech-brief/README.md (hunk 2 above)
Comment (BlueWinds) [reply packages/driver/src/cy/commands/actions/attach-file-tech-brief/README.md:57]: Exactly so. Consider the following:
```
cy.attachFile('fixtures/foo.json')

cy.readFile('fixtures/image.png', { encoding: null }).then((image) => {
  // Process image somehow
  cy.attachFile(image)
})
```

Based on how cypress-upload-file is used, I suspect these two cases will be by far the most common, and therefore wanted to make them as simple as possible, where you can pass in exactly what you want as a single argument and it "just works". That means that the same argument has to be interpreted in a couple of different ways - and since we're already parsing out what to do when *not* passed in an object, it seemed natural to use the same logic on the `contents` property when we do have an object.

Basically - option B is indeed nicer for the case when an object is passed in. But since we already have the logic for differentiating the type of the argument when it's not an object, I chose consistency.

File: packages/driver/src/cypress/error_messages.ts (hunk 1 above)
Comment (BlueWinds) [reply packages/driver/src/cypress/error_messages.ts:137]: Ah, interesting. We should consider doing this for the whole file at some point - I didn't notice that was an option because most sections don't use it.

**Review by chrisbreiding — APPROVED** [review]
